### PR TITLE
Record whether report made on desktop or mobile.

### PR DESCRIPTION
**Pull request by dracos**
Have two submit buttons, one of which displays at <48em, one of which displays at >=48em. Record desktop/mobile with report depending on which button is clicked.
If neither button is clicked, e.g. if they submit the form by hitting return (although spec says button should be clicked then? whatever), have a bit of fallback JS that sets the form field in the browser resize function.
If no-JS and no form button clicked, carry on storing nothing as now.

### CODE REVIEW

File: perllib/FixMyStreet/App/Controller/Report/New.pm [hunk 1]
```diff
@@ -1046,7 +1046,13 @@ sub process_report : Private {
     $report->detail( $detail );
 
     # mobile device type
-    $report->service( $params{service} ) if $params{service};
+    if ($c->get_param('submit_register')) {
```
Comment (zarino) [inline perllib/FixMyStreet/App/Controller/Report/New.pm:1049]: Small point, but should we handle these via the `$params` hash, as we do for everything else at this point?

**Review by zarino — APPROVED** [review]
Looks good to me.

File: perllib/FixMyStreet/App/Controller/Report/New.pm (hunk 1 above)
Comment (dracos) [reply perllib/FixMyStreet/App/Controller/Report/New.pm:1049]: You'll see that, in the main, params is for all the variables, and get_param is used for submit buttons (e.g. report_anonymously, submit_sign_in bit above). Don't think it's totally consistent, but whatever :)

**Comment by dracos** [comment]
Reordered New.pm because just realised that mobile app does use submit_register at present, so that would have then overridden the service field actually provided, so using that in preference if it's set first.

**Pull request merged**